From e5dba7fa8f6a26f8a62e61d56c451acdeffce815 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Mi=C5=82ecki?= <rafal@milecki.pl>
Date: Thu, 16 Mar 2023 19:45:58 +0530
Subject: [PATCH 125/500] OpenWrt:
 420-mtd-support-OpenWrt-s-MTD_ROOTFS_ROOT_DEV.patch
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

mtd: support OpenWrt's MTD_ROOTFS_ROOT_DEV

This allows setting ROOT_DEV to MTD partition named "rootfs".

Signed-off-by: Rafał Miłecki <rafal@milecki.pl>

Change-Id: I10fa89be5366e2064d7c28c579eef37e8bb2f9d1
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 1c7930a4348f0be3d00ee9ca6e2a68155059932a)
---
 drivers/mtd/mtdcore.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/mtdcore.c b/drivers/mtd/mtdcore.c
index 5f2f33cc3407..fa6ede33430c 100644
--- a/drivers/mtd/mtdcore.c
+++ b/drivers/mtd/mtdcore.c
@@ -768,7 +768,8 @@ int add_mtd_device(struct mtd_info *mtd)
 
 	mutex_unlock(&mtd_table_mutex);
 
-	if (of_property_read_bool(mtd_get_of_node(mtd), "linux,rootfs")) {
+	if (of_property_read_bool(mtd_get_of_node(mtd), "linux,rootfs") ||
+	+           (IS_ENABLED(CONFIG_MTD_ROOTFS_ROOT_DEV) && !strcmp(mtd->name, "rootfs") && ROOT_DEV == 0)) {
 		if (IS_BUILTIN(CONFIG_MTD)) {
 			pr_info("mtd: setting mtd%d (%s) as root device\n", mtd->index, mtd->name);
 			ROOT_DEV = MKDEV(MTD_BLOCK_MAJOR, mtd->index);
-- 
2.34.1

